src/Mvc/Mvc.Formatters.Xml/src/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -143,6 +143,8 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
                 }
 
                 readStream = new FileBufferingReadStream(request.Body, memoryThreshold);
+                // Ensure the file buffer stream is always disposed at the end of a request.
+                request.HttpContext.Response.RegisterForDispose(readStream);
```

Comment:
Looks like this completely replaces disposeReadStream.

Reply:
We continue to use `disposeReadStream` in the regular code path. It allows the buffer to be released quickly + disposes the stream asynchronously. This is in the event any other piece of code errors along the way.

Reply:
There's also RegisterForDisposeAsync

Reply:
I did not know that. I'll use that in my PR to master along with updating https://github.com/dotnet/aspnetcore/blob/4ef204e13b88c0734e0e94a1cc4c0ef05f40849e/src/Http/Http/src/Internal/BufferingHelper.cs#L26.